Tests/BugsnagOnCrashTest.m

```diff
@@ -82,16 +82,18 @@ - (void)testBlacklistedFields {
 - (void)testMergePrecedence {
     BugsnagEvent *event = [[BugsnagEvent alloc] initWithKSReport:@{
             @"user": @{
-                    @"name": @"Joe Bloggs",
+                    @"customer": @{
+                            @"name": @"Joe Bloggs",
+                    },
                     @"metaData": @{
-                            @"onCrash": @{
+                            @"customer": @{
                                     @"name": @"Beryl Merryweather",
                                     @"age": @76
                             }
                     }
             }
     }];
-    NSMutableDictionary *data = [event getMetadataFromSection:@"onCrash"];
+    NSMutableDictionary *data = [event getMetadataFromSection:@"customer"];
     XCTAssertNotNil(data);
     XCTAssertEqual(2, [data count]);
     XCTAssertEqualObjects(@"Joe Bloggs", data[@"name"]);
```

Comment:
Seems like a test for removal of a non-conformant key would be simple and worthwhile.

Reply:
Added an additional non-conforming value to this test case.